### DISCUSSION TITLE
Integrating AWS Secrets Manager with DB_CONNECTION_MUTATOR and MSSQL

### DISCUSSION THREAD

**Discussion by the-green-man14** [apache/superset#25488] [General]
Hello,

Would it be possible to implement DB_CONNECTION_MUTATOR so that a user's password and username for MSSQL database access are stored in AWS Secrets Manager (under the secret name of their LDAP username)

I would then implement the DB_CONNECTION_MUTATOR to set the db_username and db_password to the values stored in AWS Secrets Manager. 

Does this sound sensible. I am thinking something like this where I have ommited the boto3 method for talking to Secrets Manager:
def DB_CONNECTION_MUTATOR(uri, params, username, security_manager, source):
    user = security_manager.find_user(username=username)
    if user.username != "admin":
        # Retrieve the database credentials from AWS Secrets Manager
        db_username, db_password = get_database_credentials(user.username)

        if db_username and db_password:
            # Update the URI with the retrieved credentials
            uri.username = db_username
            uri.password = db_password

        # Rest of the logic for setting uri.database, uri.port, etc.
        else:
          uri.username = 'postgres'
          uri.password = 'admin_password' 

    return uri, params
    
    DB_CONNECTION_MUTATOR = 'DB_CONNECTION_MUTATOR'

**Comment by tang350476527** [comment]
谢谢您的来信，您的邮件已收到！
